sqld/src/auth.rs

```diff
@@ -119,6 +120,40 @@ impl Auth {
     }
 }
 
+impl Authenticated {
+    pub fn from_proxy_grpc<T>(req: &tonic::Request<T>) -> Result<Self, Status> {
+        let auth = match req
+            .metadata()
+            .get(GRPC_PROXY_AUTH_HEADER)
+            .map(|v| v.to_str())
+            .transpose()
+            .map_err(|_| Status::invalid_argument("missing x-proxy-auth"))?
```

Comment:
I think we should be consistent in naming here - is it `x-proxy-auth` or `x-proxy-authorization`?